diff --git a/dist/runtime/server/storage.mjs b/dist/runtime/server/storage.mjs
index 375e4bf667b777000cb9aec7ff179e4b2a530656..7a2d1041769fe0cf9e758931ff1b139b69bbee6d 100644
--- a/dist/runtime/server/storage.mjs
+++ b/dist/runtime/server/storage.mjs
@@ -56,11 +56,25 @@ export const getContentsIds = async (event, prefix) => {
   }
   return keys.filter(contentIgnorePredicate);
 };
+
+function* chunks(arr, n) {
+  for (let i = 0; i < arr.length; i += n) {
+    yield arr.slice(i, i + n)
+  }
+}
+
 export const getContentsList = async (event, prefix) => {
   const keys = await getContentsIds(event, prefix);
-  const contents = await Promise.all(keys.map((key) => getContent(event, key)));
-  return contents;
-};
+  const keyChunks = [...chunks(keys, 10)]
+  let contents = []
+  for(const chunk of keyChunks) {
+    const result = await Promise.all(chunk.map((key) => getContent(event, key)));
+    contents.push(...result)
+  }
+
+  return contents
+}
+
 export const getContent = async (event, id) => {
   const contentId = id;
   if (!contentIgnorePredicate(id)) {